server/src/main/java/io/deephaven/server/object/ObjectServiceGrpcImpl.java

```diff
@@ -26,28 +21,206 @@
+        this.errorTransformer = Objects.requireNonNull(errorTransformer);
+    }
+
+    private final class SendMessageObserver extends SingletonLivenessManager implements StreamObserver<StreamRequest> {
+
+        private ExportObject<Object> object;
+        private final SessionState session;
+        private final StreamObserver<StreamResponse> responseObserver;
+
+        private ObjectType.MessageStream messageStream;
+
+        private SendMessageObserver(SessionState session, StreamObserver<StreamResponse> responseObserver) {
+            this.session = session;
+            this.responseObserver = responseObserver;
+        }
+
+        @Override
+        public void onNext(final StreamRequest request) {
+            GrpcErrorHelper.checkHasOneOf(request, "message");
+            if (request.hasConnect()) {
+                if (this.object != null) {
+                    throw Exceptions.statusRuntimeException(Code.FAILED_PRECONDITION,
+                            "Already sent a connect request, cannot send another");
+                }
+
+                TypedTicket typedTicket = request.getConnect().getSourceId();
+
+                final String type = typedTicket.getType();
+                if (type.isEmpty()) {
+                    throw Exceptions.statusRuntimeException(Code.INVALID_ARGUMENT, "No type supplied");
+                }
+                if (typedTicket.getTicket().getTicket().isEmpty()) {
+                    throw Exceptions.statusRuntimeException(Code.INVALID_ARGUMENT, "No ticket supplied");
+                }
+
+                final SessionState.ExportObject<Object> object = ticketRouter.resolve(
+                        session, typedTicket.getTicket(), "sourceId");
+
+                this.object = object;
+                manage(this.object);
+                runOrEnqueue(() -> {
+                    final Object o = object.get();
+                    final ObjectType objectType = getObjectTypeInstance(type, o);
+
+                    PluginMessageSender clientConnection = new PluginMessageSender(responseObserver, session);
+                    messageStream = objectType.clientConnection(o, clientConnection);
+                });
+            } else if (request.hasData()) {
+                if (object == null) {
+                    throw Exceptions.statusRuntimeException(Code.INVALID_ARGUMENT,
+                            "Data message sent before Connect message");
+                }
+                Data data = request.getData();
+                List<SessionState.ExportObject<Object>> referenceObjects = data.getExportedReferencesList().stream()
+                        .map(typedTicket -> ticketRouter.resolve(session, typedTicket.getTicket(), "ticket"))
+                        .collect(Collectors.toList());
+                List<SessionState.ExportObject<Object>> requireObjects = new ArrayList<>(referenceObjects);
+                requireObjects.add(object);
```

Comment:
Looks like `runOrEnqueue` already adds requirement for `object`, so we don't need to do here?

Reply:
Correct, no longer needed.